node/pkg/driver/node_utils.go

```diff
@@ -515,3 +519,60 @@ func (n NodeUtils) GetTopologyLabels(ctx context.Context, nodeName string) (map[
 	}
 	return topologyLabels, nil
 }
+
+func (n NodeUtils) IsBlock(devicePath string) (bool, error) {
+	var stat unix.Stat_t
+	err := unix.Stat(devicePath, &stat)
+	if err != nil {
+		return false, err
+	}
+	return (stat.Mode & unix.S_IFMT) == unix.S_IFBLK, nil
+}
+
+func (d NodeUtils) GetFileSystemVolumeStats(path string) (VolumeStatistics, error) {
+	statfs := &unix.Statfs_t{}
+	err := unix.Statfs(path, statfs)
+	if err != nil {
+		return VolumeStatistics{}, err
+	}
+
+	availableBytes := int64(statfs.Bavail) * int64(statfs.Bsize)
+	totalBytes := int64(statfs.Blocks) * int64(statfs.Bsize)
+	usedBytes := (int64(statfs.Blocks) - int64(statfs.Bfree)) * int64(statfs.Bsize)
+
+	totalInodes := int64(statfs.Files)
+	availableInodes := int64(statfs.Ffree)
+	usedInodes := totalInodes - availableInodes
+
+	volumeStats := VolumeStatistics{
+		AvailableBytes: availableBytes,
+		TotalBytes:     totalBytes,
+		UsedBytes:      usedBytes,
+
+		AvailableInodes: availableInodes,
+		TotalInodes:     totalInodes,
+		UsedInodes:      usedInodes,
+	}
+
+	return volumeStats, nil
+}
+
+func (d NodeUtils) GetBlockVolumeStats(mpathDevice string) (VolumeStatistics, error) {
+	args := []string{"--getsize64", mpathDevice}
+	out, err := d.Executer.ExecuteWithTimeoutSilently(device_connectivity.TimeOutBlockDevCmd, device_connectivity.BlockDevCmd, args)
```

Comment:
why silently?

Reply:
I don't want that every that on every `grpc` call from the `kubelet` we will see all the information, it won't mess up the logs?

Reply:
it doesn't seem like a lot of information. what is the frequency we see per volume?

Reply:
about 30 seconds or less

Reply:
got it